src/features/game/toolbar/GameToolbarConstants.ts

```diff
@@ -0,0 +1,49 @@
+import ImageAssets from '../assets/ImageAssets';
+import { screenSize } from '../commons/CommonConstants';
+import { AssetKey, IBaseScene } from '../commons/CommonTypes';
+import { GamePhaseType } from '../phase/GamePhaseTypes';
+
+export type ToolbarButtonConfig = {
+  assetKey: AssetKey;
+  onUp: (scene: IBaseScene) => () => void;
+};
+
+// From rightmost to leftmost
+const buttonConfigs: ToolbarButtonConfig[] = [
+  {
+    // Escape menu button
+    assetKey: ImageAssets.gear.key,
+    onUp: scene => async () => {
+      const phaseManager = scene.getPhaseManager();
+      if (phaseManager.isCurrentPhase(GamePhaseType.EscapeMenu)) {
+        await phaseManager.popPhase();
+      } else if (phaseManager.isCurrentPhase(GamePhaseType.Dashboard)) {
```

Comment:
Perhaps it is possible to generalise GamePhaseTypes which would replace one another when called.